src/main/java/seedu/address/logic/parser/ParserUtil.java

```diff
@@ -191,6 +227,9 @@ public static Set<MedicalHistory> parseMedicals(Collection<String> medHistories)
         requireNonNull(medHistories);
         final Set<MedicalHistory> historiesSet = new HashSet<>();
         for (String historyName : medHistories) {
+            if (!MedicalHistory.isValidMedicalHistory(historyName)) {
+                throw new ParseException(MedicalHistory.MESSAGE_CONSTRAINTS);
+            }
```

Comment:
`parseMedical` in the very next step already performs this check within the method - this check isn't necessary.